main/com.spotify/dataset.tt

```diff
@@ -43,6 +43,16 @@ dataset @com.spotify language "en" {
                    "play ${p_genre:const} music"]]
     ;
 
+    program (p_song :String, p_playlist :String) := (@com.spotify.song(), id =~ p_song) => @com.spotify.add_song_to_playlist(song=id, playlist=p_playlist)
+    #_[utterances=["add ${p_song:no-undefined} to ${p_playlist:no-undefined}",
+                   "put ${p_song:no-undefined} in ${p_playlist:no-undefined}"]]
```

Comment:
It is reasonable to say "add the song XXX to some playlist" followed by the agent asking what playlist, so I think the `:no-undefined` clause is unnecessary